app/graphql/mutations/profile/edit.rb

```diff
@@ -7,22 +7,33 @@ class Edit < BaseMutation
       graphql_name 'UpdateProfile'
 
       argument :id, ID, required: true
+      argument :name, String, required: false
+      argument :description, String, required: false
       argument :compliance_threshold, Float, required: false
       argument :business_objective_id, ID, required: false
+
       field :profile, Types::Profile, null: true
 
       def resolve(args = {})
-        profile = Pundit.authorize(
-          context[:current_user],
-          ::Profile.find(args[:id]),
-          :edit?
-        )
-        profile.update(
+        profile = authorized_profile(args)
+        profile.update!(
+          name: args[:name] || profile.name,
+          description: args[:description] || profile.description,
           compliance_threshold: args[:compliance_threshold],
           business_objective_id: args[:business_objective_id]
         )
```

Comment:
If we call `update!` and pass `nil` for a field (such as `business_objective_id`), it will be updated to `nil` instead of not being updated at all.

Reply:
Yes, and in the case of `business_objective_id` we actually want that.
Not for the compliance_threshold, that is still something to be fixed.

Reply:
Not quite - make sure to support the graphql API as written, not just what the UI needs. Both `business_objective_id` and `compliance_threshold` are optional. See the graphql variables I used above in my review.